dozer/cogs/roles.py

```diff
@@ -68,6 +76,25 @@ class Roles(Cog):
             if dbrole is not None:
                 session.delete(dbrole)
 
+    async def ctxpurge(self, ctx, returncounter):
+        counter = 0
+        with db.Session() as session:
+            roles = session.query(GiveableRole).filter_by(guild_id=ctx.guild.id)
+            guildroles = []
+            for i in ctx.guild.roles:
+                guildroles.append(i.id)
+            for role in roles:
+                if role.id not in guildroles:
+                    await self.givemepurge(role)
+                    counter += 1
+        if returncounter:
```

Comment:
Because everything is truthy/falsy, you can return the counter always, and it will work for if-checks.